### ISSUE TITLE
Non descript error on attempts to unlock a post

### ISSUE THREAD

**Issue by aoduor** [ushahidi/platform#2153]
### Expected behaviour
When i click on unlock, i expect to get redirected to the edit page, or if it failed, a proper error message

### Actual behaviour
![screen capture on 2017-10-16 at 14-55-27](https://user-images.githubusercontent.com/721918/31610834-473fddf0-b282-11e7-9476-745e328195f6.gif)

### Steps to reproduce the behaviour/error
* Log into qa.ushahididev.com
* Click on data
* Select a post to edit
* Click on unlock

##### Where
- [ ] [Local setup with Vagrant ](https://www.ushahidi.com/support/install-ushahidi#installing-for-development)
- [ ] [Local setup from platform-release ](https://www.ushahidi.com/support/install-ushahidi#installing-the-latest-release)
- [ ] Ushahidi.io / SaaS solution 
- [*] Ushahidi's QA environment
- [ ] Other (explain):

**Comment by aoduor** [comment]
Also, apparently all posts on the data view on qa.ushahididev.com are locked by @rowasc 
![screen capture on 2017-10-16 at 14-59-28](https://user-images.githubusercontent.com/721918/31610932-ad9983da-b282-11e7-8e23-0a2a41bd1b10.gif)

**Comment by willdoran** [comment]
@aoduor So the post doesn't automatically redirect to the edit page.

It should however unlock. I'll look at that now.

**Comment by willdoran** [comment]
Using the QA DB and develop locally, I can't reproduce this. I'm going to clear the locks in the QA DB I think the db might just be messed up.

**Comment by willdoran** [comment]
@aoduor Ok I've cleared the DB to see if it was caused by weird state. Please retest and let me know if it persists. Something was severely screwed up because those post weren't even locked by Romina.

**Comment by aoduor** [comment]
@willdoran :(..
So, progress in the sense that, i was able to view a couple of other posts that are unlocked. Downside, i selected the very first post on the data view, and this is current status i.e i'm unable to unlock this one.

![screenshot](https://cdn-pro.dprcdn.net/files/acc_531939/D8AD3)

TL;DR: Unlocked the rest of the other posts, but cannot unlock this specific one.

**Comment by willdoran** [comment]
@aoduor I can't even get the qa site to load...

**Comment by jshorland** [comment]
I can't even get posts to show as locked anymore... am I doing something wrong to try to replicate?

**Comment by jshorland** [comment]
Ok so on sandbox.uchaguzi.or.ke, I was able to unlock a post successfully without an error message. Can someone retest this just to make sure I'm not missing something?

**Comment by willdoran** [comment]
Passes QA for me.
